### PR TITLE
Adds a highlander variant with more advanced swords

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
The advanced swords are a highlander claymore subtype with supermatter sword reactions.

This is a good idea.

:cl: Naksu
admin: Added a new variant of highlander, accessible via the secrets menu.
/:cl:

### CODE REVIEW

File: code/game/objects/items/weaponry.dm [hunk 1]
```diff
@@ -192,6 +192,46 @@
...
 	playsound(user, 'sound/items/screwdriver2.ogg', 50, 1)
 
+/obj/item/claymore/highlander/supermatter // THIS IS THE BEST IDEA
+	var/obj/machinery/power/supermatter_shard/shard
+
+/obj/item/claymore/highlander/supermatter/Initialize()
+	. = ..()
+	shard = new(src)
+	qdel(shard.countdown)
+	shard.countdown = null
```
Comment (Cyberboss) [inline code/game/objects/items/weaponry.dm]: QDEL_NULL

File: code/game/objects/items/weaponry.dm [hunk 2]
```diff
@@ -192,6 +192,46 @@
...
+
+/obj/item/claymore/highlander/supermatter/afterattack(target, mob/user, proximity_flag)
+	..()
+	if(user && target == user)
+		user.dropItemToGround(src)
```
Comment (Cyberboss) [inline code/game/objects/items/weaponry.dm]: , TRUE for force

File: code/game/objects/items/weaponry.dm [hunk 3]
```diff
@@ -192,6 +192,46 @@
...
+/obj/item/claymore/highlander/supermatter/proc/consume_turf(turf/T)
+	var/oldtype = T.type
+	var/turf/newT = T.ScrapeAway()
+	if(newT.type == oldtype)
+		return
+	playsound(T, 'sound/effects/supermatter.ogg', 50, 1)
+	T.visible_message("<span class='danger'>[T] smacks into [src] and rapidly flashes to ash.</span>",\
+	"<span class='italics'>You hear a loud crack as you are washed with a wave of heat.</span>")
+	shard.Consume()
+	T.CalculateAdjacentTurfs()
```
Comment (Cyberboss) [inline code/game/objects/items/weaponry.dm]: ScrapeAway should handle this

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Cyberboss** [comment]
Changelog

**Review by Cyberboss — DISMISSED** [review]

**Comment by DaxDupont** [comment]
for what purpose is this [s]

**Comment by CitrusGender** [comment]
This was tested and people didn't like it that much.

It was done on high-pop though, so it's to be expected.
Kinda made holes in the station and wasn't as good as normal highlander.
It was alright but I can tell other people didn't like it.

**Comment by AlexTheSysop42** [comment]
Make the swords not emit radiation, also maybe add something to keep you from stabbing yourself by accident

**Comment by Xhuis** [comment]
This is pretty half-baked. It doesn't have any nuke disk mechanics, which makes the mode impossible to win, and doesn't account for the shard's radiation which would end up killing the wielder and everyone near them. Highlander is already pretty fine as-is, and I see this as something that's used once in a blue moon because the actual highlander is more enjoyable.

File: code/modules/admin/verbs/onlyone.dm [hunk 4]
```diff
@@ -25,5 +25,8 @@ GLOBAL_VAR_INIT(highlander, FALSE)
...
 	addtimer(CALLBACK(src, .proc/only_one), 420)
 
-/mob/living/carbon/human/proc/make_scottish()
-	mind.add_antag_datum(/datum/antagonist/highlander)
\ No newline at end of file
+/mob/living/carbon/human/proc/make_scottish(super = FALSE)
+	if (super)
+		mind.add_antag_datum(/datum/antagonist/highlander/supermatter)
+	else
+		mind.add_antag_datum(/datum/antagonist/highlander)
```
Comment (coiax) [inline code/modules/admin/verbs/onlyone.dm]: Have you tested if this works correctly with latejoins?

Comment (AlexTheSysop42) [reply code/modules/admin/verbs/onlyone.dm]: it doesnt work with ghost roles for sure

**Comment by CitrusGender** [comment]
*inhale* god yes

**Comment by CitrusGender** [comment]
*inhales deeper*

**Comment by CitrusGender** [comment]
results: as expected. absolutely hated

**Comment by AlexTheSysop42** [comment]
Newjoiners get a normal highlander sword instead of a supermatter one and get vaporized instantly

**Comment by birchbark** [comment]
Let be explain why this gamemode is shit.

The combat in SS13 is already pretty simplistic, with some depth of gameplay in it's ability to manipulate with grabs, disarms, and targeting specific body parts. The problem is, this gamemode exclusively relies on "Who clicks first wins", and it takes 15 minutes for this literally braindead combat to eventually shred the station down to a single person. It's a 'who has lower ping contest' in the worst way. 

It's the literal incarnation of murderboner, in all it's retarded and simplistic incarnations. Space station 13 is not defined exclusively by it's combat, not even by powergaming fucktards who are the reason we have the term 'shitcurity'. Implementation of this gamemode in a more reasonable way would be some kind of bartender game where anyone can challenge a head to a duel, and obtain their privilege through some badge of honor. Alternatively, make it a shuttle you can buy on an emagged console, where it only allows for two players in some kind of combat arena. Go play a stupid fucking battle royale in a game where combat has more depth than just clicking on someone until they fall over. Jesus christ.

**Comment by CitrusGender** [comment]
@Xhuis same

**Pull request closed without merging**